src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -379,6 +379,7 @@ void ImageReplayer<I>::bootstrap() {
 
   {
     Mutex::Locker locker(m_lock);
+    request->get();
```

Comment:
Because explicitly deleting does not work for unit tests (mock objects). Note, the request objects are always created by `create()` static method, and see how it is overrode in mock, e.g: https://github.com/ceph/ceph/blob/master/src/test/rbd_mirror/test_mock_ImageReplayer.cc#L55

Another solution would be to provide `destroy()` static method and use it instead of explicitly deleting.